packages/web-components/fast-foundation/src/tree-item/tree-item.ts

```diff
@@ -287,7 +287,6 @@ export class TreeItem extends FASTElement {
     }
 
     private handleSelected(e?: Event): void {
-        this.selected = !this.selected;
         this.$emit("selected-change", e);
```

Comment:
It seems the item will still emit a "selected-change" event even though the selected state isn't actually changing, is that accurate? If so, that seems wrong to me. It should only emit a change event if it is actually changing

Reply:
I think that makes sense - I think clarity of where we want to handle selection will help. The selected state here was somewhat duplicative to the logic which intended to prevent change in the event the same tree item was selected. Changing here requires us to know about the previous selected state in comparison to other items. Not firing a change event here means that the click and keydown events would move to the tree. I'm fine with that but would love confirmation that folks are good with that shift before I rework.

Reply:
I totally agree, focus navigation is split between the tree and the items as well.

In the future I'm starting to think that the Tree should handle all focus navigation *and* selection because:
1. Event delegation makes consolidating event handlers to a single element more efficient
2. Reduce coordinating logic between items and elements
3.  Replacing items with custom item implementations becomes easier because business logic is consolidated.

If that is something we want to aim for, I think we could move this piece now by removing the click handler from the item and having the tree manage selection. *setting* selected would emit a `selected-change' event but the tree-view would disregard the event because it already handled selection; the event would just be for authors.

Reply:
Maybe rename the event to something like "item-invoked" would make it clear that the item is not controlling it's own selected state?

Reply:
I think I would consider *renaming* an event a breaking change. To me (and this is up for debate), the emitted events could be relied upon but authors and therefore part of the public "API"

Reply:
Yes, I'd agree that it would be a breaking change.

Reply:
Revisiting this - do we have a desired path for the change here? A couple of proposals; to be honest, I'm open.

Reply:
@nicholasrice pinging again - I agree we don't want to necessarily rename here as part of this. Curious if we still feel like the above proposed solution is the right way to address this.

Reply:
I still believe so, yes.